examples/multiphysics/rising_bubble_VOF/rising_bubble.py

```diff
@@ -0,0 +1,118 @@
+#############################################################################
+"""
+Postprocessing code for rising_bubble_VOF example
+
+"""
+#############################################################################
+
+'''Importing Libraries'''
+from math import pi
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import matplotlib.tri as tri
+from matplotlib import cm
+import pyvista as pv
+from scipy.signal import savgol_filter
+
+import glob
+
+import os
+import sys
+from pyvista.plotting.renderer import CameraPosition
+
+from scipy.linalg.special_matrices import dft
+#############################################################################
+
+#############################################################################
+'''Functions'''
+
+#############################################################################
+
+#Take case path as argument and store at currentPath
+currentPath = sys.argv[1]
+
+#Define list of VTK files and time list:
+list_vtu = os.listdir(currentPath)
+list_vtu = [x for x in list_vtu if "pvd" not in x ]
+list_vtu = [x for x in list_vtu if "pvtu" not in x ]
+
+#Create a list of time_steps
+data = pd.read_csv("Output/rising_bubble_VOF.pvd",sep='"',header=5, usecols=[1]) 
+data.columns = ["a"] 
+time_list = data['a']
+print(data['a'])
+
+#Set phase_limit to search for maximum x
+phase_limit = 0.5
+
+#Sort list
+time_list, list_vtu = (list(t) for t in zip(*sorted(zip(time_list, list_vtu))))
+
+#Create lists to fill with average y, dy, dt and bubble rise velocity
+y_list = []
+y_diff_list = []
+t_diff_list = []
+bubble_rise_vel = []
+
+#Read vtu data
+for i in range(0, len(list_vtu)):
+    #Read DF from VTK files
+    exec(f'df_{i} = pv.read(f\'{currentPath}/{list_vtu[i]}\')')
+
+    #Select a data to apply the slice   
+    exec(f'df = df_{i}')
+
+    #find average 'y' by averaging minimum and maximum y of the bubble
+    points = pd.DataFrame(df.points[:, 1])
+    points_x = pd.DataFrame(df.points[:, 0])
+    phase  = pd.DataFrame(df['phase'])
+    
+    y_max = max(points[phase[0] > phase_limit].values)[0]
+    y_min = min(points[phase[0] > phase_limit].values)[0]
+    y_mean = (y_min + y_max) * 0.5
+    y_list.append(y_mean)
+       
+#Print('List of average values for y:')
+print(y_list)
+
+#Sort y_list, since the time_list is extracted from the .pvd file, the y_list and time_list are not consistent
+y_list.sort()
+
+#Smooth average y
+smoothed_y = savgol_filter(y_list, 7, 2)
+
+#Calculate bubble rise velocity = dy/dt
+y_diff_list = np.diff(smoothed_y)
+t_diff_list = np.diff(time_list)
+bubble_rise_vel = y_diff_list / t_diff_list
+bubble_rise_t = time_list[0:-1] + t_diff_list
```

Comment:
I would use a second ordered centered derrivative
Also, why not derive the smoothed instead of the raw y? It would make a lot more sense IMO.

Reply:
applied.